### PR TITLE
remove redundant post_id and thread_id

### PR DESCRIPTION
**Pull request by juniuszhou**
remove redundant post_id and thread_id

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -1319,10 +1313,10 @@ decl_module! {
             Self::ensure_is_forum_member_with_correct_account(&who, &forum_user_id)?;
 
             // Add new post
-            let post = Self::add_new_post(thread_id, &text, forum_user_id)?;
+            Self::add_new_post(thread_id, &text, forum_user_id)?;
 
             // Generate event
-            Self::deposit_event(RawEvent::PostAdded(post.id));
+            Self::deposit_event(RawEvent::PostAdded(<NextPostId<T>>::get() - One::one()));
```
Comment (bedeho) [inline src/lib.rs]: Simplify, separate out computation from posting event.

File: src/lib.rs [hunk 2]
```diff
@@ -1394,7 +1388,7 @@ decl_module! {
             });
 
             // Generate event
-            Self::deposit_event(RawEvent::PostTextUpdated(post.id, <PostById<T>>::get(post_id).text_change_history.len() as u64));
+            Self::deposit_event(RawEvent::PostTextUpdated(post_id, <PostById<T>>::get(post_id).text_change_history.len() as u64));
```
Comment (bedeho) [inline src/lib.rs]: Simplify, separate out writing to storage from posting event.

**Review by bedeho — CHANGES_REQUESTED** [review]

File: src/lib.rs (hunk 1 above)
Comment (juniuszhou) [reply src/lib.rs]: accepted.

File: src/lib.rs (hunk 2 above)
Comment (juniuszhou) [reply src/lib.rs]: accepted.

**Review by bedeho — APPROVED** [review]

**Pull request merged**